src/gui/application.cpp

```diff
@@ -66,6 +66,72 @@
 
 class QSocket;
 
+namespace {
+
+void migrateConfigFile(const QCoreApplication *app)
+{
+    using namespace OCC;
+
+    if (!ConfigFile::exists()) {
+        // check whether an old config location must be migrated
+        // we support multiple locations from old versions
+        // these are worked on in-order to upgrade from version to version
+        // the algorithm is the same for all these locations, thus we can use a loop
+        QStringList configLocationsToMigrate;
+
+        // note: this change is temporary to allow using QDesktopServices etc. to determine the paths
+        // the application name was changed to
+        const auto oldApplicationName = app->applicationName();
+        QCoreApplication::setApplicationName(Theme::instance()->appNameGUI());
+
+        // location used in versions <= 2.4
+        // We need to use the deprecated QDesktopServices::storageLocation because of its Qt4 behavior of adding "data" to the path
+        configLocationsToMigrate.append(QDesktopServices::storageLocation(QDesktopServices::DataLocation));
+
+        // location used in versions from 2.5 to 2.8
+        configLocationsToMigrate.append(QStandardPaths::writableLocation(Utility::isWindows() ? QStandardPaths::AppDataLocation : QStandardPaths::AppConfigLocation));
+
+        // reset to original value
+        app->setApplicationName(oldApplicationName);
+
+        for (auto oldDir : configLocationsToMigrate) {
+            if (oldDir.endsWith('/')) {
+                // macOS 10.11.x does not like trailing slash for rename/move.
+                oldDir.chop(1);
+            }
+
+            if (QFileInfo(oldDir).isDir()) {
+                auto confDir = ConfigFile::configPath();
+                if (confDir.endsWith('/'))
+                    confDir.chop(1); // macOS 10.11.x does not like trailing slash for rename/move.
+                qCInfo(lcApplication) << "Migrating old config from" << oldDir << "to" << confDir;
+
+                if (!QFile::rename(oldDir, confDir)) {
+                    qCWarning(lcApplication) << "Failed to move the old config directory to its new location (" << oldDir << "to" << confDir << ")";
+
+                    // Try to move the files one by one
+                    if (QFileInfo(confDir).isDir() || QDir().mkdir(confDir)) {
```

Comment:
While touching this, maybe https://doc.qt.io/qt-5/qdir.html#entryInfoList-1 instead of manually constructing the path.

`QDir().mkdir(confDir)` will try to create `confDir` in the current working dir 😲 I'd say that never worked.